### PR TITLE
Recursive implementation of Bubble sort

### PR DESCRIPTION
**Pull request by Krishnapal4050**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1240"></a>

### CODE REVIEW

**Comment by Krishnapal4050** [comment]
@kvedala As suggested by you, when I rebased that repository, that pull request got closed automatically. So, I had to create a new PR after rebasing this branch.
Please review it now and give your valuable suggestions.

**Comment by kvedala** [comment]
👍 
Please review the contribution guidelines. Especially: https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program
and update the code before the review. Thank you

**Comment by kvedala** [comment]
You can also review the existing code: https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/sorting/insertion_sort.cpp for reference.

File: sorting/recursive_bubble_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      { 
+          std::swap(arr[i], arr[i+1]); //swap elements
+      } 
+   }
+
+   recBubble(arr, n-1);
+}
+
+
+int main()
```
Comment (Panquesito7) [inline sorting/recursive_bubble_sort.cpp:57]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main()
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: DIRECTORY.md [hunk 2]
```diff
@@ -113,7 +113,9 @@
   * [Adaline Learning](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/machine_learning/adaline_learning.cpp)
   * [Kohonen Som Topology](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/machine_learning/kohonen_som_topology.cpp)
   * [Kohonen Som Trace](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/machine_learning/kohonen_som_trace.cpp)
+  * [Neural Network](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/machine_learning/neural_network.cpp)
```
Comment (aadijay3477) [inline DIRECTORY.md:116]: Needs a review i think it's irrelevant.

Comment (Panquesito7) [reply DIRECTORY.md:116]: Those changes were made by the GitHub Actions itself.
AFAIK, if we revert those changes, they will be made automatically again.

File: sorting/recursive_bubble_sort.cpp (hunk 1 above)
Comment (Panquesito7) [reply sorting/recursive_bubble_sort.cpp:57]: Not resolved/fixed yet.

**Pull request closed without merging**